### DISCUSSION TITLE
Unable to apply transform on Mednist dataset while implementing 2D image registration

### DISCUSSION THREAD

**Discussion by kitu-datta** [Project-MONAI/MONAI#2085] [Q&A]
Hi ,

I would like to say that on implementing learning based image registration (2D) notebook as mentioned below
https://github.com/Project-MONAI/tutorials/blob/master/2d_registration/registration_mednist.ipynb, 
I see I am unable to use transform.
I get "RuntimeError: applying transform <monai.transforms.compose.Compose object at 0x7fe3fc41dd50>" when I am trying to run below cell as shown in screenshot
![Capture](https://user-images.githubusercontent.com/77524540/115983437-af660480-a5be-11eb-9e34-e3afd2be34ba.PNG)

Where as I was able to load input data successfully
![Capture1](https://user-images.githubusercontent.com/77524540/115983551-7a0de680-a5bf-11eb-80c5-88677b16dbbc.PNG)

Would appreciate your help here.
Thanks

**Comment by wyli** [comment]
the current notebook works fine on colab https://colab.research.google.com/github/Project-MONAI/tutorials/blob/master/2d_registration/registration_mednist.ipynb

could you include some info of what you modified and the detailed error messages?

**Comment by kitu-datta** [comment]
Hi @wyli ,

It is indeed working fine in colab.
However, I am just trying to run the notebook in my EC2 instance. I observed, while executing below cell I get "**ERROR: Could not build wheels for monai which use PEP 517 and cannot be installed directly**" 
![Capture2](https://user-images.githubusercontent.com/77524540/116106720-00b1e900-a6d0-11eb-9dfc-7b47762541be.PNG)
 it seems it is unable to build MONAI's C++/CUDA extensions. I have attached the notebook here with error logs , in zipped file. 

[Image_Registration.zip](https://github.com/Project-MONAI/MONAI/files/6378564/Image_Registration.zip)



Hope this helps.
Thanks

**Reply by wyli** [reply]
I see, you can replace that cell with something like `!pip install monai`  which doesn't use the c++/CUDA ext, but the pipeline should also work

**Reply by kitu-datta** [reply]
Hi @wyli,

I figured out that I was having problem with the previous kernel. 
The notebook works fine in my new environment with  `!pip install monai.` 
But is there any alternative way to build extensions.
I tried first ` pip install pep517` then `BUILD_MONAI=1 pip install git+https://github.com/Project-MONAI/MONAI#egg=monai[all]` in my new environment, still I get the same error

"**Failed to build monai
ERROR: Could not build wheels for monai which use PEP 517 and cannot be installed directly**"

If I am not wrong , on building extensions MONAI  would have preferred its own extensions instead of the Pytorch/Python native implementations, when calling some relevant python modules

**Reply by wyli** [reply]
there is an alternative method to install, please follow https://docs.monai.io/en/latest/installation.html#option-2-editable-installation . currently the extensions are experimental, more rigorous evaluations are still needed
